hibernate-core/src/main/java/org/hibernate/query/sqm/tree/expression/SqmSelfRenderingExpression.java

```diff
@@ -51,4 +51,6 @@ public <X> X accept(SemanticQueryWalker<X> walker) {
 	public void appendHqlString(StringBuilder hql, SqmRenderContext context) {
 		throw new UnsupportedOperationException();
 	}
+
+	//TODO: what is a correct impl of equals() / hashCode() here?
```

Comment:
Since the whole state that defines this node is in the `renderer`, I would argue that checking equality on the renderer is the way to go, though AFAICS, this stuff is only ever used internally and is irrelevant for caching, so basing equality on the object identity is just as fine.